### ISSUE TITLE
Bash auto-complete stuck for at least 10 seconds when press tab from within deepin-terminal

### ISSUE THREAD

**Issue by happyd0ggie** [linuxdeepin/developer-center#10725]
### SUMMARY | 问题概要

When I press tab from within deepni-terminal to let bash-completion complete the remaining part of file name or directory name, it stuck for at least 10 seconds.

### SOFTWARE/OS VERSIONS | 软件/系统版本信息

Distribution: Deepin V23 insider testing channel
Packages:
- deepin-terminal 6.0.14
- bash-completion 1:2.11-5
- expect 5.45.4-deepin1

### PRE-REQUIREMENT | 预置条件

Upgrade system to 2024/9/30 updates.

### STEPS TO REPRODUCE | 复现步骤

1. Open deepin-terminal
2. Input directory prefix two chars that exists
3. Press tab to let bash-completion work

### OBSERVED RESULT | 观察到的结果

The bash-completion stuck for at least 10 seconds to complete.

### EXPECTED RESULT | 期望的结果

The bash-completion completes instantly.

**Comment by happyd0ggie** [comment]
I am using [starship](https://starship.rs/) 1.20.1, there is a relevant issue: [some commands when executed freezes for around 10 seconds before showing Error: i/o timeout and finishing. · Issue #6057 · starship/starship](https://github.com/starship/starship/issues/6057)

**Comment by luodeepin** [comment]
Hello!
We are not reproducing the TAB not working when entering commands in the terminal, do you have the problem only when entering in the triple tool starship?

**Comment by happyd0ggie** [comment]
> Hello! We are not reproducing the TAB not working when entering commands in the terminal, do you have the problem only when entering in the triple tool starship?

Not really. The problem persists after exit starship. So I assume the problem is not starship relevant.

FYI, starship is loading from `~/.bashrc`, there is one line `eval "$(starship init bash)"` in `~/.bashrc` file.

BWT, is this issue relevant to [首次执行sudo 需要等待10秒钟左右－论坛－深度科技](https://bbs.deepin.org/zh/post/279415?offset=0&postId=1652689)

**Comment by luodeepin** [comment]
Hello! Would it be possible for you to open a remote at your convenience on Thursday or Friday, and I'll arrange for a developer to look at the situation for you on Thursday or Friday?

**Comment by happyd0ggie** [comment]
Contact me @wechat 18202702570
---- Replied Message ----
From ***@***.***> Date 10/15/2024 13:55 To ***@***.***> Cc ***@***.***>***@***.***> Subject Re: [linuxdeepin/developer-center] Bash auto-complete stuck for at least 10 seconds when press tab from within deepin-terminal (Issue #10725) 
Hello! Would it be possible for you to open a remote at your convenience on Thursday or Friday, and I'll arrange for a developer to look at the situation for you on Thursday or Friday?

**Comment by happyd0ggie** [comment]
> Hello! Would it be possible for you to open a remote at your convenience on Thursday or Friday, and I'll arrange for a developer to look at the situation for you on Thursday or Friday?

I can provide remote control of my machine between 13:30-14:00 this Friday. Please provide your email address, and I will send the remove control to you.

**Comment by luodeepin** [comment]
> > Hello! Would it be possible for you to open a remote at your convenience on Thursday or Friday, and I'll arrange for a developer to look at the situation for you on Thursday or Friday?
> 
> I can provide remote control of my machine between 13:30-14:00 this Friday. Please provide your email address, and I will send the remove control to you.

Hello! My email is luoxinsi@uniontech.com
